### DISCUSSION TITLE
When Clause for Webview View

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#642] [Extension Development QnA] [answered]
I want to show my webview view only when my file explorer contains a specific file at any path in explorer. I tried resoucreFilename but only works if that file is open in editior. What is the right way to achieve this?

**Comment by r3m0t** [comment, marked as answer]
Invent a new context name, set it in the extension JS and read it in package.json

**Comment by alefragnani** [comment]
Hi @amitjoshi438 ,

Do you mean, whenever you open a folder/workspace that _contains some specific file in it_ your WebView should be automatically opened? If this is the case, take a look at **Activation Event**. There is one, called `workspaceContains` where you can define a glob pattern for the desired file. More details in https://code.visualstudio.com/api/references/activation-events#workspaceContains.

On the other hand, if you need to open the webview when _some specific file is created within the folder/workspace_, then you could simply create a `FileWatcher` to detect the file being created inside the folder/workspace you have opened, and when created, shows the WebView.

Hope this helps
